// @aws-sdk/client-bedrock/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  Bedrock: () => Bedrock,
  BedrockClient: () => BedrockClient,
  BedrockServiceException: () => BedrockServiceException,
  CommitmentDuration: () => CommitmentDuration,
  ConflictException: () => ConflictException,
  CreateModelCustomizationJobCommand: () => CreateModelCustomizationJobCommand,
  CreateProvisionedModelThroughputCommand: () => CreateProvisionedModelThroughputCommand,
  CustomizationType: () => CustomizationType,
  DeleteCustomModelCommand: () => DeleteCustomModelCommand,
  DeleteModelInvocationLoggingConfigurationCommand: () => DeleteModelInvocationLoggingConfigurationCommand,
  DeleteProvisionedModelThroughputCommand: () => DeleteProvisionedModelThroughputCommand,
  FineTuningJobStatus: () => FineTuningJobStatus,
  FoundationModelLifecycleStatus: () => FoundationModelLifecycleStatus,
  GetCustomModelCommand: () => GetCustomModelCommand,
  GetFoundationModelCommand: () => GetFoundationModelCommand,
  GetModelCustomizationJobCommand: () => GetModelCustomizationJobCommand,
  GetModelInvocationLoggingConfigurationCommand: () => GetModelInvocationLoggingConfigurationCommand,
  GetProvisionedModelThroughputCommand: () => GetProvisionedModelThroughputCommand,
  InferenceType: () => InferenceType,
  InternalServerException: () => InternalServerException,
  ListCustomModelsCommand: () => ListCustomModelsCommand,
  ListFoundationModelsCommand: () => ListFoundationModelsCommand,
  ListModelCustomizationJobsCommand: () => ListModelCustomizationJobsCommand,
  ListProvisionedModelThroughputsCommand: () => ListProvisionedModelThroughputsCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ModelCustomization: () => ModelCustomization,
  ModelCustomizationJobStatus: () => ModelCustomizationJobStatus,
  ModelModality: () => ModelModality,
  ProvisionedModelStatus: () => ProvisionedModelStatus,
  PutModelInvocationLoggingConfigurationCommand: () => PutModelInvocationLoggingConfigurationCommand,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  SortByProvisionedModels: () => SortByProvisionedModels,
  SortJobsBy: () => SortJobsBy,
  SortModelsBy: () => SortModelsBy,
  SortOrder: () => SortOrder,
  StopModelCustomizationJobCommand: () => StopModelCustomizationJobCommand,
  TagResourceCommand: () => TagResourceCommand,
  ThrottlingException: () => ThrottlingException,
  TooManyTagsException: () => TooManyTagsException,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateProvisionedModelThroughputCommand: () => UpdateProvisionedModelThroughputCommand,
  ValidationException: () => ValidationException,
  __Client: () => import_smithy_client.Client,
  paginateListCustomModels: () => paginateListCustomModels,
  paginateListModelCustomizationJobs: () => paginateListModelCustomizationJobs,
  paginateListProvisionedModelThroughputs: () => paginateListProvisionedModelThroughputs
});
module.exports = __toCommonJS(src_exports);

// src/BedrockClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "bedrock"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/BedrockClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/BedrockClient.ts
var _BedrockClient = class _BedrockClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultBedrockHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_BedrockClient, "BedrockClient");
var BedrockClient = _BedrockClient;

// src/Bedrock.ts


// src/commands/CreateModelCustomizationJobCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");

// src/models/BedrockServiceException.ts

var _BedrockServiceException = class _BedrockServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _BedrockServiceException.prototype);
  }
};
__name(_BedrockServiceException, "BedrockServiceException");
var BedrockServiceException = _BedrockServiceException;

// src/models/models_0.ts
var _AccessDeniedException = class _AccessDeniedException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var _InternalServerException = class _InternalServerException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ThrottlingException = class _ThrottlingException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var _ValidationException = class _ValidationException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var _ConflictException = class _ConflictException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var CustomizationType = {
  CONTINUED_PRE_TRAINING: "CONTINUED_PRE_TRAINING",
  FINE_TUNING: "FINE_TUNING"
};
var ModelCustomization = {
  CONTINUED_PRE_TRAINING: "CONTINUED_PRE_TRAINING",
  FINE_TUNING: "FINE_TUNING"
};
var InferenceType = {
  ON_DEMAND: "ON_DEMAND",
  PROVISIONED: "PROVISIONED"
};
var ModelModality = {
  EMBEDDING: "EMBEDDING",
  IMAGE: "IMAGE",
  TEXT: "TEXT"
};
var FoundationModelLifecycleStatus = {
  ACTIVE: "ACTIVE",
  LEGACY: "LEGACY"
};
var SortModelsBy = {
  CREATION_TIME: "CreationTime"
};
var SortOrder = {
  ASCENDING: "Ascending",
  DESCENDING: "Descending"
};
var CommitmentDuration = {
  ONE_MONTH: "OneMonth",
  SIX_MONTHS: "SixMonths"
};
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var _TooManyTagsException = class _TooManyTagsException extends BedrockServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TooManyTagsException",
      $fault: "client",
      ...opts
    });
    this.name = "TooManyTagsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TooManyTagsException.prototype);
    this.resourceName = opts.resourceName;
  }
};
__name(_TooManyTagsException, "TooManyTagsException");
var TooManyTagsException = _TooManyTagsException;
var ProvisionedModelStatus = {
  CREATING: "Creating",
  FAILED: "Failed",
  IN_SERVICE: "InService",
  UPDATING: "Updating"
};
var SortByProvisionedModels = {
  CREATION_TIME: "CreationTime"
};
var ModelCustomizationJobStatus = {
  COMPLETED: "Completed",
  FAILED: "Failed",
  IN_PROGRESS: "InProgress",
  STOPPED: "Stopped",
  STOPPING: "Stopping"
};
var SortJobsBy = {
  CREATION_TIME: "CreationTime"
};
var FineTuningJobStatus = {
  COMPLETED: "Completed",
  FAILED: "Failed",
  IN_PROGRESS: "InProgress",
  STOPPED: "Stopped",
  STOPPING: "Stopping"
};

// src/protocols/Aws_restJson1.ts
var se_CreateModelCustomizationJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/model-customization-jobs");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      baseModelIdentifier: [],
      clientRequestToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      customModelKmsKeyId: [],
      customModelName: [],
      customModelTags: (_) => (0, import_smithy_client._json)(_),
      customizationType: [],
      hyperParameters: (_) => (0, import_smithy_client._json)(_),
      jobName: [],
      jobTags: (_) => (0, import_smithy_client._json)(_),
      outputDataConfig: (_) => (0, import_smithy_client._json)(_),
      roleArn: [],
      trainingDataConfig: (_) => (0, import_smithy_client._json)(_),
      validationDataConfig: (_) => (0, import_smithy_client._json)(_),
      vpcConfig: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateModelCustomizationJobCommand");
var se_CreateProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/provisioned-model-throughput");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientRequestToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      commitmentDuration: [],
      modelId: [],
      modelUnits: [],
      provisionedModelName: [],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateProvisionedModelThroughputCommand");
var se_DeleteCustomModelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/custom-models/{modelIdentifier}");
  b.p("modelIdentifier", () => input.modelIdentifier, "{modelIdentifier}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteCustomModelCommand");
var se_DeleteModelInvocationLoggingConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/logging/modelinvocations");
  let body;
  body = "";
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteModelInvocationLoggingConfigurationCommand");
var se_DeleteProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/provisioned-model-throughput/{provisionedModelId}");
  b.p("provisionedModelId", () => input.provisionedModelId, "{provisionedModelId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteProvisionedModelThroughputCommand");
var se_GetCustomModelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/custom-models/{modelIdentifier}");
  b.p("modelIdentifier", () => input.modelIdentifier, "{modelIdentifier}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetCustomModelCommand");
var se_GetFoundationModelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/foundation-models/{modelIdentifier}");
  b.p("modelIdentifier", () => input.modelIdentifier, "{modelIdentifier}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetFoundationModelCommand");
var se_GetModelCustomizationJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/model-customization-jobs/{jobIdentifier}");
  b.p("jobIdentifier", () => input.jobIdentifier, "{jobIdentifier}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetModelCustomizationJobCommand");
var se_GetModelInvocationLoggingConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/logging/modelinvocations");
  let body;
  body = "";
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetModelInvocationLoggingConfigurationCommand");
var se_GetProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/provisioned-model-throughput/{provisionedModelId}");
  b.p("provisionedModelId", () => input.provisionedModelId, "{provisionedModelId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetProvisionedModelThroughputCommand");
var se_ListCustomModelsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/custom-models");
  const query = (0, import_smithy_client.map)({
    [_cTB]: [
      () => input.creationTimeBefore !== void 0,
      () => (input[_cTB].toISOString().split(".")[0] + "Z").toString()
    ],
    [_cTA]: [
      () => input.creationTimeAfter !== void 0,
      () => (input[_cTA].toISOString().split(".")[0] + "Z").toString()
    ],
    [_nC]: [, input[_nC]],
    [_bMAE]: [, input[_bMAE]],
    [_fMAE]: [, input[_fMAE]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]],
    [_sB]: [, input[_sB]],
    [_sO]: [, input[_sO]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListCustomModelsCommand");
var se_ListFoundationModelsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/foundation-models");
  const query = (0, import_smithy_client.map)({
    [_bP]: [, input[_bP]],
    [_bCT]: [, input[_bCT]],
    [_bOM]: [, input[_bOM]],
    [_bIT]: [, input[_bIT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListFoundationModelsCommand");
var se_ListModelCustomizationJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/model-customization-jobs");
  const query = (0, import_smithy_client.map)({
    [_cTA]: [
      () => input.creationTimeAfter !== void 0,
      () => (input[_cTA].toISOString().split(".")[0] + "Z").toString()
    ],
    [_cTB]: [
      () => input.creationTimeBefore !== void 0,
      () => (input[_cTB].toISOString().split(".")[0] + "Z").toString()
    ],
    [_sE]: [, input[_sE]],
    [_nC]: [, input[_nC]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]],
    [_sB]: [, input[_sB]],
    [_sO]: [, input[_sO]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListModelCustomizationJobsCommand");
var se_ListProvisionedModelThroughputsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/provisioned-model-throughputs");
  const query = (0, import_smithy_client.map)({
    [_cTA]: [
      () => input.creationTimeAfter !== void 0,
      () => (input[_cTA].toISOString().split(".")[0] + "Z").toString()
    ],
    [_cTB]: [
      () => input.creationTimeBefore !== void 0,
      () => (input[_cTB].toISOString().split(".")[0] + "Z").toString()
    ],
    [_sE]: [, input[_sE]],
    [_mAE]: [, input[_mAE]],
    [_nC]: [, input[_nC]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]],
    [_sB]: [, input[_sB]],
    [_sO]: [, input[_sO]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListProvisionedModelThroughputsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/listTagsForResource");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      resourceARN: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_PutModelInvocationLoggingConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/logging/modelinvocations");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      loggingConfig: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_PutModelInvocationLoggingConfigurationCommand");
var se_StopModelCustomizationJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/model-customization-jobs/{jobIdentifier}/stop");
  b.p("jobIdentifier", () => input.jobIdentifier, "{jobIdentifier}", false);
  let body;
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StopModelCustomizationJobCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tagResource");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      resourceARN: [],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/untagResource");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      resourceARN: [],
      tagKeys: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/provisioned-model-throughput/{provisionedModelId}");
  b.p("provisionedModelId", () => input.provisionedModelId, "{provisionedModelId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      desiredModelId: [],
      desiredProvisionedModelName: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateProvisionedModelThroughputCommand");
var de_CreateModelCustomizationJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    jobArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateModelCustomizationJobCommand");
var de_CreateProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    provisionedModelArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateProvisionedModelThroughputCommand");
var de_DeleteCustomModelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteCustomModelCommand");
var de_DeleteModelInvocationLoggingConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteModelInvocationLoggingConfigurationCommand");
var de_DeleteProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteProvisionedModelThroughputCommand");
var de_GetCustomModelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    baseModelArn: import_smithy_client.expectString,
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    customizationType: import_smithy_client.expectString,
    hyperParameters: import_smithy_client._json,
    jobArn: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    modelArn: import_smithy_client.expectString,
    modelKmsKeyArn: import_smithy_client.expectString,
    modelName: import_smithy_client.expectString,
    outputDataConfig: import_smithy_client._json,
    trainingDataConfig: import_smithy_client._json,
    trainingMetrics: (_) => de_TrainingMetrics(_, context),
    validationDataConfig: import_smithy_client._json,
    validationMetrics: (_) => de_ValidationMetrics(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetCustomModelCommand");
var de_GetFoundationModelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    modelDetails: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetFoundationModelCommand");
var de_GetModelCustomizationJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    baseModelArn: import_smithy_client.expectString,
    clientRequestToken: import_smithy_client.expectString,
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    customizationType: import_smithy_client.expectString,
    endTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    failureMessage: import_smithy_client.expectString,
    hyperParameters: import_smithy_client._json,
    jobArn: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    lastModifiedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    outputDataConfig: import_smithy_client._json,
    outputModelArn: import_smithy_client.expectString,
    outputModelKmsKeyArn: import_smithy_client.expectString,
    outputModelName: import_smithy_client.expectString,
    roleArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    trainingDataConfig: import_smithy_client._json,
    trainingMetrics: (_) => de_TrainingMetrics(_, context),
    validationDataConfig: import_smithy_client._json,
    validationMetrics: (_) => de_ValidationMetrics(_, context),
    vpcConfig: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetModelCustomizationJobCommand");
var de_GetModelInvocationLoggingConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    loggingConfig: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetModelInvocationLoggingConfigurationCommand");
var de_GetProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    commitmentDuration: import_smithy_client.expectString,
    commitmentExpirationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    desiredModelArn: import_smithy_client.expectString,
    desiredModelUnits: import_smithy_client.expectInt32,
    failureMessage: import_smithy_client.expectString,
    foundationModelArn: import_smithy_client.expectString,
    lastModifiedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    modelArn: import_smithy_client.expectString,
    modelUnits: import_smithy_client.expectInt32,
    provisionedModelArn: import_smithy_client.expectString,
    provisionedModelName: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetProvisionedModelThroughputCommand");
var de_ListCustomModelsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    modelSummaries: (_) => de_CustomModelSummaryList(_, context),
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListCustomModelsCommand");
var de_ListFoundationModelsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    modelSummaries: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListFoundationModelsCommand");
var de_ListModelCustomizationJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    modelCustomizationJobSummaries: (_) => de_ModelCustomizationJobSummaries(_, context),
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListModelCustomizationJobsCommand");
var de_ListProvisionedModelThroughputsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    provisionedModelSummaries: (_) => de_ProvisionedModelSummaries(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListProvisionedModelThroughputsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_PutModelInvocationLoggingConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_PutModelInvocationLoggingConfigurationCommand");
var de_StopModelCustomizationJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_StopModelCustomizationJobCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateProvisionedModelThroughputCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateProvisionedModelThroughputCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.bedrock#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.bedrock#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.bedrock#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.bedrock#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.bedrock#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.bedrock#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "TooManyTagsException":
    case "com.amazonaws.bedrock#TooManyTagsException":
      throw await de_TooManyTagsExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.bedrock#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(BedrockServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var de_TooManyTagsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new TooManyTagsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_TooManyTagsExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var de_CustomModelSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    baseModelArn: import_smithy_client.expectString,
    baseModelName: import_smithy_client.expectString,
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    customizationType: import_smithy_client.expectString,
    modelArn: import_smithy_client.expectString,
    modelName: import_smithy_client.expectString
  });
}, "de_CustomModelSummary");
var de_CustomModelSummaryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_CustomModelSummary(entry, context);
  });
  return retVal;
}, "de_CustomModelSummaryList");
var de_ModelCustomizationJobSummaries = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ModelCustomizationJobSummary(entry, context);
  });
  return retVal;
}, "de_ModelCustomizationJobSummaries");
var de_ModelCustomizationJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    baseModelArn: import_smithy_client.expectString,
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    customModelArn: import_smithy_client.expectString,
    customModelName: import_smithy_client.expectString,
    customizationType: import_smithy_client.expectString,
    endTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    jobArn: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    lastModifiedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    status: import_smithy_client.expectString
  });
}, "de_ModelCustomizationJobSummary");
var de_ProvisionedModelSummaries = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ProvisionedModelSummary(entry, context);
  });
  return retVal;
}, "de_ProvisionedModelSummaries");
var de_ProvisionedModelSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    commitmentDuration: import_smithy_client.expectString,
    commitmentExpirationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    desiredModelArn: import_smithy_client.expectString,
    desiredModelUnits: import_smithy_client.expectInt32,
    foundationModelArn: import_smithy_client.expectString,
    lastModifiedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    modelArn: import_smithy_client.expectString,
    modelUnits: import_smithy_client.expectInt32,
    provisionedModelArn: import_smithy_client.expectString,
    provisionedModelName: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_ProvisionedModelSummary");
var de_TrainingMetrics = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    trainingLoss: import_smithy_client.limitedParseFloat32
  });
}, "de_TrainingMetrics");
var de_ValidationMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ValidatorMetric(entry, context);
  });
  return retVal;
}, "de_ValidationMetrics");
var de_ValidatorMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    validationLoss: import_smithy_client.limitedParseFloat32
  });
}, "de_ValidatorMetric");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var _bCT = "byCustomizationType";
var _bIT = "byInferenceType";
var _bMAE = "baseModelArnEquals";
var _bOM = "byOutputModality";
var _bP = "byProvider";
var _cTA = "creationTimeAfter";
var _cTB = "creationTimeBefore";
var _fMAE = "foundationModelArnEquals";
var _mAE = "modelArnEquals";
var _mR = "maxResults";
var _nC = "nameContains";
var _nT = "nextToken";
var _sB = "sortBy";
var _sE = "statusEquals";
var _sO = "sortOrder";

// src/commands/CreateModelCustomizationJobCommand.ts
var _CreateModelCustomizationJobCommand = class _CreateModelCustomizationJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "CreateModelCustomizationJob", {}).n("BedrockClient", "CreateModelCustomizationJobCommand").f(void 0, void 0).ser(se_CreateModelCustomizationJobCommand).de(de_CreateModelCustomizationJobCommand).build() {
};
__name(_CreateModelCustomizationJobCommand, "CreateModelCustomizationJobCommand");
var CreateModelCustomizationJobCommand = _CreateModelCustomizationJobCommand;

// src/commands/CreateProvisionedModelThroughputCommand.ts




var _CreateProvisionedModelThroughputCommand = class _CreateProvisionedModelThroughputCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "CreateProvisionedModelThroughput", {}).n("BedrockClient", "CreateProvisionedModelThroughputCommand").f(void 0, void 0).ser(se_CreateProvisionedModelThroughputCommand).de(de_CreateProvisionedModelThroughputCommand).build() {
};
__name(_CreateProvisionedModelThroughputCommand, "CreateProvisionedModelThroughputCommand");
var CreateProvisionedModelThroughputCommand = _CreateProvisionedModelThroughputCommand;

// src/commands/DeleteCustomModelCommand.ts




var _DeleteCustomModelCommand = class _DeleteCustomModelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "DeleteCustomModel", {}).n("BedrockClient", "DeleteCustomModelCommand").f(void 0, void 0).ser(se_DeleteCustomModelCommand).de(de_DeleteCustomModelCommand).build() {
};
__name(_DeleteCustomModelCommand, "DeleteCustomModelCommand");
var DeleteCustomModelCommand = _DeleteCustomModelCommand;

// src/commands/DeleteModelInvocationLoggingConfigurationCommand.ts




var _DeleteModelInvocationLoggingConfigurationCommand = class _DeleteModelInvocationLoggingConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "DeleteModelInvocationLoggingConfiguration", {}).n("BedrockClient", "DeleteModelInvocationLoggingConfigurationCommand").f(void 0, void 0).ser(se_DeleteModelInvocationLoggingConfigurationCommand).de(de_DeleteModelInvocationLoggingConfigurationCommand).build() {
};
__name(_DeleteModelInvocationLoggingConfigurationCommand, "DeleteModelInvocationLoggingConfigurationCommand");
var DeleteModelInvocationLoggingConfigurationCommand = _DeleteModelInvocationLoggingConfigurationCommand;

// src/commands/DeleteProvisionedModelThroughputCommand.ts




var _DeleteProvisionedModelThroughputCommand = class _DeleteProvisionedModelThroughputCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "DeleteProvisionedModelThroughput", {}).n("BedrockClient", "DeleteProvisionedModelThroughputCommand").f(void 0, void 0).ser(se_DeleteProvisionedModelThroughputCommand).de(de_DeleteProvisionedModelThroughputCommand).build() {
};
__name(_DeleteProvisionedModelThroughputCommand, "DeleteProvisionedModelThroughputCommand");
var DeleteProvisionedModelThroughputCommand = _DeleteProvisionedModelThroughputCommand;

// src/commands/GetCustomModelCommand.ts




var _GetCustomModelCommand = class _GetCustomModelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "GetCustomModel", {}).n("BedrockClient", "GetCustomModelCommand").f(void 0, void 0).ser(se_GetCustomModelCommand).de(de_GetCustomModelCommand).build() {
};
__name(_GetCustomModelCommand, "GetCustomModelCommand");
var GetCustomModelCommand = _GetCustomModelCommand;

// src/commands/GetFoundationModelCommand.ts




var _GetFoundationModelCommand = class _GetFoundationModelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "GetFoundationModel", {}).n("BedrockClient", "GetFoundationModelCommand").f(void 0, void 0).ser(se_GetFoundationModelCommand).de(de_GetFoundationModelCommand).build() {
};
__name(_GetFoundationModelCommand, "GetFoundationModelCommand");
var GetFoundationModelCommand = _GetFoundationModelCommand;

// src/commands/GetModelCustomizationJobCommand.ts




var _GetModelCustomizationJobCommand = class _GetModelCustomizationJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "GetModelCustomizationJob", {}).n("BedrockClient", "GetModelCustomizationJobCommand").f(void 0, void 0).ser(se_GetModelCustomizationJobCommand).de(de_GetModelCustomizationJobCommand).build() {
};
__name(_GetModelCustomizationJobCommand, "GetModelCustomizationJobCommand");
var GetModelCustomizationJobCommand = _GetModelCustomizationJobCommand;

// src/commands/GetModelInvocationLoggingConfigurationCommand.ts




var _GetModelInvocationLoggingConfigurationCommand = class _GetModelInvocationLoggingConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "GetModelInvocationLoggingConfiguration", {}).n("BedrockClient", "GetModelInvocationLoggingConfigurationCommand").f(void 0, void 0).ser(se_GetModelInvocationLoggingConfigurationCommand).de(de_GetModelInvocationLoggingConfigurationCommand).build() {
};
__name(_GetModelInvocationLoggingConfigurationCommand, "GetModelInvocationLoggingConfigurationCommand");
var GetModelInvocationLoggingConfigurationCommand = _GetModelInvocationLoggingConfigurationCommand;

// src/commands/GetProvisionedModelThroughputCommand.ts




var _GetProvisionedModelThroughputCommand = class _GetProvisionedModelThroughputCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "GetProvisionedModelThroughput", {}).n("BedrockClient", "GetProvisionedModelThroughputCommand").f(void 0, void 0).ser(se_GetProvisionedModelThroughputCommand).de(de_GetProvisionedModelThroughputCommand).build() {
};
__name(_GetProvisionedModelThroughputCommand, "GetProvisionedModelThroughputCommand");
var GetProvisionedModelThroughputCommand = _GetProvisionedModelThroughputCommand;

// src/commands/ListCustomModelsCommand.ts




var _ListCustomModelsCommand = class _ListCustomModelsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "ListCustomModels", {}).n("BedrockClient", "ListCustomModelsCommand").f(void 0, void 0).ser(se_ListCustomModelsCommand).de(de_ListCustomModelsCommand).build() {
};
__name(_ListCustomModelsCommand, "ListCustomModelsCommand");
var ListCustomModelsCommand = _ListCustomModelsCommand;

// src/commands/ListFoundationModelsCommand.ts




var _ListFoundationModelsCommand = class _ListFoundationModelsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "ListFoundationModels", {}).n("BedrockClient", "ListFoundationModelsCommand").f(void 0, void 0).ser(se_ListFoundationModelsCommand).de(de_ListFoundationModelsCommand).build() {
};
__name(_ListFoundationModelsCommand, "ListFoundationModelsCommand");
var ListFoundationModelsCommand = _ListFoundationModelsCommand;

// src/commands/ListModelCustomizationJobsCommand.ts




var _ListModelCustomizationJobsCommand = class _ListModelCustomizationJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "ListModelCustomizationJobs", {}).n("BedrockClient", "ListModelCustomizationJobsCommand").f(void 0, void 0).ser(se_ListModelCustomizationJobsCommand).de(de_ListModelCustomizationJobsCommand).build() {
};
__name(_ListModelCustomizationJobsCommand, "ListModelCustomizationJobsCommand");
var ListModelCustomizationJobsCommand = _ListModelCustomizationJobsCommand;

// src/commands/ListProvisionedModelThroughputsCommand.ts




var _ListProvisionedModelThroughputsCommand = class _ListProvisionedModelThroughputsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "ListProvisionedModelThroughputs", {}).n("BedrockClient", "ListProvisionedModelThroughputsCommand").f(void 0, void 0).ser(se_ListProvisionedModelThroughputsCommand).de(de_ListProvisionedModelThroughputsCommand).build() {
};
__name(_ListProvisionedModelThroughputsCommand, "ListProvisionedModelThroughputsCommand");
var ListProvisionedModelThroughputsCommand = _ListProvisionedModelThroughputsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "ListTagsForResource", {}).n("BedrockClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/PutModelInvocationLoggingConfigurationCommand.ts




var _PutModelInvocationLoggingConfigurationCommand = class _PutModelInvocationLoggingConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "PutModelInvocationLoggingConfiguration", {}).n("BedrockClient", "PutModelInvocationLoggingConfigurationCommand").f(void 0, void 0).ser(se_PutModelInvocationLoggingConfigurationCommand).de(de_PutModelInvocationLoggingConfigurationCommand).build() {
};
__name(_PutModelInvocationLoggingConfigurationCommand, "PutModelInvocationLoggingConfigurationCommand");
var PutModelInvocationLoggingConfigurationCommand = _PutModelInvocationLoggingConfigurationCommand;

// src/commands/StopModelCustomizationJobCommand.ts




var _StopModelCustomizationJobCommand = class _StopModelCustomizationJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "StopModelCustomizationJob", {}).n("BedrockClient", "StopModelCustomizationJobCommand").f(void 0, void 0).ser(se_StopModelCustomizationJobCommand).de(de_StopModelCustomizationJobCommand).build() {
};
__name(_StopModelCustomizationJobCommand, "StopModelCustomizationJobCommand");
var StopModelCustomizationJobCommand = _StopModelCustomizationJobCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "TagResource", {}).n("BedrockClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "UntagResource", {}).n("BedrockClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateProvisionedModelThroughputCommand.ts




var _UpdateProvisionedModelThroughputCommand = class _UpdateProvisionedModelThroughputCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockControlPlaneService", "UpdateProvisionedModelThroughput", {}).n("BedrockClient", "UpdateProvisionedModelThroughputCommand").f(void 0, void 0).ser(se_UpdateProvisionedModelThroughputCommand).de(de_UpdateProvisionedModelThroughputCommand).build() {
};
__name(_UpdateProvisionedModelThroughputCommand, "UpdateProvisionedModelThroughputCommand");
var UpdateProvisionedModelThroughputCommand = _UpdateProvisionedModelThroughputCommand;

// src/Bedrock.ts
var commands = {
  CreateModelCustomizationJobCommand,
  CreateProvisionedModelThroughputCommand,
  DeleteCustomModelCommand,
  DeleteModelInvocationLoggingConfigurationCommand,
  DeleteProvisionedModelThroughputCommand,
  GetCustomModelCommand,
  GetFoundationModelCommand,
  GetModelCustomizationJobCommand,
  GetModelInvocationLoggingConfigurationCommand,
  GetProvisionedModelThroughputCommand,
  ListCustomModelsCommand,
  ListFoundationModelsCommand,
  ListModelCustomizationJobsCommand,
  ListProvisionedModelThroughputsCommand,
  ListTagsForResourceCommand,
  PutModelInvocationLoggingConfigurationCommand,
  StopModelCustomizationJobCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateProvisionedModelThroughputCommand
};
var _Bedrock = class _Bedrock extends BedrockClient {
};
__name(_Bedrock, "Bedrock");
var Bedrock = _Bedrock;
(0, import_smithy_client.createAggregatedClient)(commands, Bedrock);

// src/pagination/ListCustomModelsPaginator.ts

var paginateListCustomModels = (0, import_core.createPaginator)(BedrockClient, ListCustomModelsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListModelCustomizationJobsPaginator.ts

var paginateListModelCustomizationJobs = (0, import_core.createPaginator)(BedrockClient, ListModelCustomizationJobsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListProvisionedModelThroughputsPaginator.ts

var paginateListProvisionedModelThroughputs = (0, import_core.createPaginator)(BedrockClient, ListProvisionedModelThroughputsCommand, "nextToken", "nextToken", "maxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  BedrockServiceException,
  __Client,
  BedrockClient,
  Bedrock,
  $Command,
  CreateModelCustomizationJobCommand,
  CreateProvisionedModelThroughputCommand,
  DeleteCustomModelCommand,
  DeleteModelInvocationLoggingConfigurationCommand,
  DeleteProvisionedModelThroughputCommand,
  GetCustomModelCommand,
  GetFoundationModelCommand,
  GetModelCustomizationJobCommand,
  GetModelInvocationLoggingConfigurationCommand,
  GetProvisionedModelThroughputCommand,
  ListCustomModelsCommand,
  ListFoundationModelsCommand,
  ListModelCustomizationJobsCommand,
  ListProvisionedModelThroughputsCommand,
  ListTagsForResourceCommand,
  PutModelInvocationLoggingConfigurationCommand,
  StopModelCustomizationJobCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateProvisionedModelThroughputCommand,
  paginateListCustomModels,
  paginateListModelCustomizationJobs,
  paginateListProvisionedModelThroughputs,
  AccessDeniedException,
  InternalServerException,
  ThrottlingException,
  ValidationException,
  ConflictException,
  ResourceNotFoundException,
  CustomizationType,
  ModelCustomization,
  InferenceType,
  ModelModality,
  FoundationModelLifecycleStatus,
  SortModelsBy,
  SortOrder,
  CommitmentDuration,
  ServiceQuotaExceededException,
  TooManyTagsException,
  ProvisionedModelStatus,
  SortByProvisionedModels,
  ModelCustomizationJobStatus,
  SortJobsBy,
  FineTuningJobStatus
});
